tests/testthat/test-istr.R

```diff
@@ -41,7 +41,6 @@ test_that("if `companies` lacks crucial columns, errors gracefully", {
   expect_error(istr(bad, scenarios, inputs), crucial)
 })
 
-# TODO: Common test in both istr and pstr because both use same scenario dataset
```

Comment:
@kalashsinghal, is this the blocking change you request or did I miss it?

Reply:
@maurolepore you are right! this test should stay in both PSTR and ISTR. However, this test will become identical after changes through this issue #367. Maybe after having an `xstr_scenario_dataset`, we can create a separate `test-scenario_dataset.R` file for such common scenario specific tests! Hence, feel free to update/delete this TODO according to your best understanding.

Also want to flag that this is not a blocking change I requested

Reply:
OK, I think I now see the source of our misunderstanding. 

You seem interested in testing the properties of the example dataset `scenarios` rather than the properties of the functions `pstr()` and `istr()`. If that 's the case then I agree we need a test in test-xstr_scenarios.R -- or in tests/testthat/test-data.R that we already have. 

My argument is for tests about the functions, so I think it's OK to keep one test for each function even if they use the same data. 

To illustrate my point consider these two toy functions. They use the same data and are similar (like `pstr()` and `istr()`) but not identical so they would behave differently under test.

``` r
data <- tibble::tibble(x = 1)

f <- function(data) {
  data
}

g <- function(data) {
  stop("This is a bug!")
  data
}

# f() and g() use the same data but they potentially behave differently so each
# needs its own test

f(data)
#> # A tibble: 1 × 1
#>       x
#>   <dbl>
#> 1     1

g(data)
#> Error in g(data): This is a bug!
```

<sup>Created on 2023-06-01 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
When hitting "Request changes" during review merge is blocked. 

![image](https://github.com/2DegreesInvesting/tiltIndicator/assets/5856545/635745e8-cd8f-4247-8781-709d48b68ee5)


But from your comment above I understand you're OK with me merging so I"ll bypass the block and merge using my admin privileges.
![image](https://github.com/2DegreesInvesting/tiltIndicator/assets/5856545/15d5545e-1028-4e87-bfc0-f77ee3bf7f6f)

Reply:
@maurolepore I agree :)